libraries/botbuilder-core/botbuilder/core/Teams/teams_activity_handler.py

```diff
@@ -0,0 +1,189 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+from http import HTTPStatus
+from botbuilder.schema import ActivityTypes, ChannelAccount
+from botbuilder.core.turn_context import TurnContext
+from botbuilder.core import ActivityHandler, MessageFactory, InvokeResponse
+from botbuilder.schema.teams import (
+    TeamInfo,
+    ChannelInfo,
+    TeamsChannelData,
+    TeamsChannelAccount,
+)
+from botframework.connector import Channels
+
+class TeamsActivityHandler(ActivityHandler):
+    async def on_turn(self, turn_context: TurnContext):
+        if turn_context is None:
+            raise TypeError("ActivityHandler.on_turn(): turn_context cannot be None.")
+
+        if hasattr(turn_context, "activity") and turn_context.activity is None:
+            raise TypeError(
+                "ActivityHandler.on_turn(): turn_context must have a non-None activity."
+            )
+
+        if (
+            hasattr(turn_context.activity, "type")
+            and turn_context.activity.type is None
+        ):
+            raise TypeError(
+                "ActivityHandler.on_turn(): turn_context activity must have a non-None type."
+            )
+
+        if turn_context.activity.type == ActivityTypes.invoke:
+            invoke_response = await self.on_invoke_activity(turn_context)
+        else:
+            await super().on_turn(turn_context)
+
+    async def on_invoke_activity(self, turn_context: TurnContext):
+        try:
+            if (
+                turn_context.activity.name is None
+                and turn_context.activity.channel_id == Channels.Msteams
+            ):
+                return  # await on_teams_card_action_invoke_activity(turn_context)
+
+            turn_context.send_activity(MessageFactory.text("working"))
+        except:
+            return
+
+    async def on_conversation_update_activity(self, turn_context: TurnContext):
+        if turn_context.activity.channel_id == Channels.ms_teams:
+            channel_data = TeamsChannelData(**turn_context.activity.channel_data)
+
+            if turn_context.activity.members_added is not None:
```

Comment:
same concept here, it's better to change it to `if turn_context.activity.members_added` to cover falsey cases